File: hw/fuentes/U3/HW19ABSTRACT/HWABSTRACT/ec.edu.espe.client_AbstractFactory_jar_1.0-SNAPSHOT/src/main/java/ec/edu/espe/client/model/GUIFactory.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.client.model;

/**
 *
 * @author jorge
 */
public abstract class GUIFactory {
    public static GUIFactory getFactory(){
        int sys = readFromConfigFile("OS_TYPE");
        if(sys==0)
            return(new WinFactory());
        else
            return(new LinuxFactory());
    }
    
    private static int readFromConfigFile(String key) {
        String os = System.getProperty("os.name").toLowerCase();
        if (os.contains("win")) {
            return 0;
        } else {
            return 1;
        }

    }
    
    public abstract Button createButton();
    public abstract Menu createeMenu();
    
}

File: hw/jacome/U3/HW19AbstractFactory/JS/controller/GUIFactory.js
class GUIFactory {
    static getFactory() {
        const os = require('os').platform().toLowerCase();
        if (os.includes('win')) {
            return new (require('./WinFactory'))();
        } else {
            return new (require('./LinuxFactory'))();
        }
    }

    createButton() {
        throw new Error("Abstract method createButton() must be implemented");
    }

    createMenu() {
        throw new Error("Abstract method createMenu() must be implemented");
    }
}

module.exports = GUIFactory;